app/policies/accounting/transaction_policy.rb

```diff
@@ -1,6 +1,7 @@
 class Accounting::TransactionPolicy < ApplicationPolicy
   def index?
-    machine_user_or_appropriate_division_admin?
+    member_level_access = user && user != :machine && user.accessible_division_ids.include?(division.id)
```

Comment:
Two things here:

1. It seems a bit off to me to have the User model responsible for determining which divisions it has access to. Seems like a violation of SRP. One approach would be to just spell out `DivisionPolicy::Scope.new(user, Division).accessible_ids`. But even that, in a way, seems a bit odd—get a list of IDs of divisions the user has access to and then check if this division is in it. Should that logic just be encapsulated within DivisionPolicy? I'd expect to see something link `DivisionPolicy.new(user, division).access?`. (It appears `access?` is different than `show?`, please confirm this is correct.) Could we try that, and later (not now) try to phase out the `accessible_division` methods from `User`?

2. To match the general style of Policy classes, I think you could put this line in a private method just like `machine_user_or_appropriate_division_admin?` and then you wouldn't need the local var.